collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -704,13 +701,19 @@ public void run() {
                 break;
             case AUDIO_CAPTURE:
             case VIDEO_CAPTURE:
-                Uri mediaUri = intent.getData();
-                saveAudioVideoAnswer(mediaUri);
-                String filePath = MediaUtils.getDataColumn(this, mediaUri, null, null);
-                if (filePath != null) {
-                    new File(filePath).delete();
+                try{
+                    Uri mediaUri = intent.getData();
+                    saveAudioVideoAnswer(mediaUri);
+                    String filePath = MediaUtils.getDataColumn(this, mediaUri, null, null);
+                    if (filePath != null) {
+                        new File(filePath).delete();
+                    }
+                    getContentResolver().delete(mediaUri, null, null);
+                }
+                catch (Exception e){
+                    Toast.makeText(this, "Error recording Audio, Please try again", Toast.LENGTH_SHORT).show();
```

Comment:
Here trying again wont help. Because that particular app's return intent cannot be well handled. It wont change itself.